> the version fix is correct, but `sync_rate_table` is gated on the same `rates_changed` flag. When only a rate `description` changes, the JSON blob saves the new value but the `Rate` table row keeps the old one.
> 
> you can verify this in the `test_update_metadata_does_not_increment_version` unittest to confirm the desync.
> 
> ```python
> def test_update_metadata_does_not_increment_version(self):
>     """Test that metadata-only updates do not increment version."""
>     # ...
> 
>     update_data["rates"][0]["description"] = "updated rate description"
>     response = self.client.put(detail_url, data=update_data, format="json", **self.headers)
>     self.assertEqual(response.status_code, status.HTTP_200_OK)
>     self.assertEqual(response.data["version"], 1)
> 
>     # verify the Rate table row is also updated 
>     with tenant_context(self.tenant):
>         rate = Rate.objects.get(price_list__uuid=pl_uuid)
>         self.assertEqual(rate.description, "updated rate description")
> ```
> 
> adding an assertion for the description results in test failure
> 
> ```
> ======================================================================
> FAIL: test_update_metadata_does_not_increment_version (cost_models.test.test_price_list_view.PriceListViewTests.test_update_metadata_does_not_increment_version)
> Test that metadata-only updates do not increment version.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/koku/koku/cost_models/test/test_price_list_view.py", line 211, in test_update_metadata_does_not_increment_version
>     self.assertEqual(rate.description, "updated rate description")
> AssertionError: '' != 'updated rate description'
> + updated rate description
> ```

fixed

--- a/koku/cost_models/test/test_price_list_view.py
+++ b/koku/cost_models/test/test_price_list_view.py
@@ -14,6 +14,7 @@ from api.iam.test.iam_test_case import IamTestCase
 from cost_models.models import CostModel
 from cost_models.models import PriceList
 from cost_models.models import PriceListCostModelMap
+from cost_models.models import Rate
 
 
 class PriceListViewTests(IamTestCase):
@@ -203,6 +204,10 @@ class PriceListViewTests(IamTestCase):
         response = self.client.put(detail_url, data=update_data, format="json", **self.headers)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.data["version"], 1)
+
+        with tenant_context(self.tenant):
+            rate = Rate.objects.get(price_list__uuid=pl_uuid)
+            self.assertEqual(rate.description, "updated rate description")
 
     def test_update_currency_fails(self):
         """Test that updating currency fails - currency is immutable."""

